src/providers/base_javascript.js

```diff
@@ -1,212 +1,339 @@
+			)
+		);
+	}
+
+	/**
+   * Executes the list command to get dependencies
+   * @param {boolean} includeTransitive - Whether to include transitive dependencies
+   * @param {string} manifestDir - The manifest directory
+   * @returns {string} The command output
+   * @private
+   */
 	#executeListCmd(includeTransitive, manifestDir) {
 		const listArgs = this._listCmdArgs(includeTransitive, manifestDir);
 		return this.#invokeCommand(listArgs);
 	}
 
-	#version() {
-		this.#invokeCommand(['--version'], { stdio: 'ignore' });
+	/**
+   * Gets the version of the package manager
+   * @returns {string} The version string of the package manager
+   * @protected
+   */
+	_version() {
+		return this.#invokeCommand(['--version']);
 	}
 
+	/**
+   * Creates or updates the lock file for the package manager
+   * @param {string} manifestDir - Directory containing the manifest file
+   * @private
+   */
 	#createLockFile(manifestDir) {
-		// in windows os, --prefix flag doesn't work, it behaves really weird , instead of installing the package.json fromm the prefix folder,
-		// it's installing package.json (placed in current working directory of process) into prefix directory, so
-		let originalDir = process.cwd()
-		if (os.platform() === 'win32') {
-			process.chdir(manifestDir)
+		const originalDir = process.cwd();
+		const isWindows = os.platform() === 'win32';
+
+		if (isWindows) {
+			// On Windows, --prefix flag doesn't work as expected
+			// Instead of installing from the prefix folder, it installs from current working directory
+			process.chdir(manifestDir);
 		}
-		const args = this._updateLockFileCmdArgs(manifestDir);
+
 		try {
+			const args = this._updateLockFileCmdArgs(manifestDir);
 			this.#invokeCommand(args);
 		} finally {
-			if (os.platform() === 'win32') {
-				process.chdir(originalDir)
+			if (isWindows) {
+				process.chdir(originalDir);
 			}
 		}
 	}
 
+	/**
+   * Invokes a command with the given arguments
+   * @param {string[]} args - Command arguments
+   * @param {Object} [opts={}] - Optional configuration options
+   * @returns {string} Command output
+   * @throws {Error} If command execution fails or command is not found
+   * @private
+   */
 	#invokeCommand(args, opts = {}) {
 		try {
+			if(!opts.cwd) {
+				opts.cwd = path.dirname(this.#manifest.manifestPath);
+			}
 			return invokeCommand(this.#cmd, args, opts);
 		} catch (error) {
 			if (error.code === 'ENOENT') {
-				throw new Error(`${this.#cmd} is not accessible`);
+				throw new Error(`${this.#cmd} is not accessible. Please ensure it is installed and available in your PATH.`);
 			}
-			throw new Error(`failed to execute ${this.#cmd} ${args}`, { cause: error })
+			if (error.code === 'EACCES') {
+				throw new Error(`Permission denied when executing ${this.#cmd}. Please check file permissions.`);
+			}
+			throw new Error(`Failed to execute ${this.#cmd} ${args.join(' ')}: ${error.message}`, { cause: error });
```

Comment:
We'll already be including `cause.message` at the top-level once I open the PR containing [this change](https://github.com/Strum355/fabric8-analytics-vscode-extension/pull/1/files#diff-39b2554fd18da165b59a6351b1aafff3714e2a80c1435f2de9706355b4d32351R26)

Reply:
Updated